### PR TITLE
FIX: Use selected text in the link if text is already selected, otherwise, put the URL

### PR DESCRIPTION
**Pull request by cpradio**

### CODE REVIEW

**Comment by eviltrout** [comment]
This should really have a test in `d-editor-test.js.es6`

**Comment by cpradio** [comment]
Test added :smile:

File: test/javascripts/components/d-editor-test.js.es6 [hunk 1]
```diff
@@ -193,6 +193,23 @@ testCase('link modal (simple link)', function(assert) {
   });
 });
 
+testCase('link modal (simple link) with selected text', function(assert, textarea) {
+  this.set('value', "hello world.");
```
Comment (eviltrout) [inline test/javascripts/components/d-editor-test.js.es6]: If you are using the `testCase` helper it should default to `hello world.`. Also you shouldn't need an `andThen` to set the selectionStart and End, other tests don't  need it!

Comment (cpradio) [inline test/javascripts/components/d-editor-test.js.es6]: https://github.com/cpradio/discourse/blob/cbde168281e2c7a37a0a4701677d8a0e0ec32344/test/javascripts/components/d-editor-test.js.es6#L98-L101

This uses andThen to select text...

**Comment by cpradio** [comment]
Updated and rebased to clean up the commits

**Comment by eviltrout** [comment]
Thanks!

**Pull request merged**